core/include/traccc/clusterization/details/sparse_ccl.hpp

```diff
@@ -42,25 +42,29 @@ TRACCC_HOST_DEVICE inline unsigned int make_union(
 
 /// Helper method to find adjacent cells
 ///
-/// @param a the first cell
-/// @param b the second cell
+/// @param cells All silicon cells in the event
+/// @param i     The index of the first cell
+/// @param j     The index of the second cell
 ///
 /// @return boolan to indicate 8-cell connectivity
 ///
-TRACCC_HOST_DEVICE inline bool is_adjacent(const traccc::cell& a,
-                                           const traccc::cell& b);
+TRACCC_HOST_DEVICE inline bool is_adjacent(
+    const edm::silicon_cell_collection::const_device& cells, unsigned int i,
+    unsigned int j);
```

Comment:
This is a nice example of a place where a proxy object would be _really_ nice; complicating this function signature and making it so much less general is a bit unfortunate.

Reply:
Have a look if you like the latest version of the code. :wink: I only made the clusterization code make use of `vecmem::edm::proxy` for now, just to see how everyone likes it.